apis/accesscontextmanager/v1beta1/accesslevel_identity.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common/identity"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// AccessLevelIdentityURL is the format for the externalRef of an AccessPolicy.
+	// This format is used in the AccessContextManager API (and is different.
+	// See: https://docs.cloud.google.com/asset-inventory/docs/asset-names
+	AccessLevelIdentityURL = "accessPolicies/{{accessPolicyID}}/accessLevels/{{accessLevel}}"
+)
+
+var _ identity.Identity = &AccessLevelIdentity{}
+var _ identity.Resource = &AccessContextManagerAccessLevel{}
+
+var parser = regexp.MustCompile(`((//)?accesscontextmanager.googleapis.com)?/?accessPolicies/(?P<accessPolicies>[[:alpha:]]+)/accessLevels/(?P<accessLevels>[[:alpha:]]+)`)
+
+// AccessLevelIdentity represents the identity of an accessLevel.
+// +k8s:deepcopy-gen=false
+type AccessLevelIdentity struct {
+	Parent      string
+	AccessLevel string
+}
+
+func (i *AccessLevelIdentity) String() string {
+	return "accessPolicies/" + i.Parent + "/accessLevels/" + i.AccessLevel
+}
+
+func (i *AccessLevelIdentity) FromExternal(ref string) error {
+	// TODO: Should be able to parse https://docs.cloud.google.com/asset-inventory/docs/asset-names
+	// But that format is //cloudresourcemanager.googleapis.com/accessPolicieis/ACCESS_POLICY_ID/accessLevels/ACCESS_LEVEL
+	// which is not the format used by the service.
+
+	err, identityMap := parseIdentityMap(ref, 2)
+	if err != nil {
+		return fmt.Errorf("format of AccessLevel external=%q was not known (use %s): %w", ref, AccessLevelIdentityURL, err)
+	}
+
+	i.Parent = identityMap["accessPolicies"]
+	i.AccessLevel = identityMap["accessLevels"]
+
+	return nil
+}
+
+func parseIdentityMap(ref string, cnt int) (error, map[string]string) {
+	raw := parser.FindStringSubmatch(ref)
```

Comment:
Can we / should we match the whole string?

Reply:
(Not a blocker, but I feel like we should think about this because lots of URLs are prefixes of other URLs)

Reply:
I think it depends on why we are parsing out the pieces. If we the goal is just to determine the resource if form then I would use something more like 
```
regexp.MustCompile(`((//)?accesscontextmanager.googleapis.com)?/?(?P<resourceID>accessPolicies/[[:alpha:]]+/accessLevels/[[:alpha:]]+)`)
```
However I suspect with some resources we may want to get at least some of the individual fields out ('project id' comes to mind). Given that I was thinking it made sense to just extract each of the individual fields.